### PR TITLE
workaround for invalid references (WP-772)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Helpers/MetaFieldProcessor/ReferencedStdBasedContentProcessorAbstract.php [hunk 1]
```diff
@@ -80,7 +77,7 @@ public function processFieldPostTranslation(SubmissionEntity $submission, $field
         }
 
         $attachment = $this->submissionManager->findOne([
-            SubmissionEntity::FIELD_CONTENT_TYPE => ContentTypeHelper::POST_TYPE_ATTACHMENT,
+            SubmissionEntity::FIELD_CONTENT_TYPE => $this->detectRealContentType($submission->getSourceBlogId(), $value),
```
Comment (vsolovei-smartling) [inline inc/Smartling/Helpers/MetaFieldProcessor/ReferencedStdBasedContentProcessorAbstract.php:80]: Part one of the fix, assuming we always work with attachments in this step was wrong

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 2]
```diff
@@ -583,7 +584,7 @@ public function getRelations(string $contentType, int $id, array $targetBlogIds)
                     foreach ($ids as $detectedId) {
                         if ($detectedId === null) {
                             $this->getLogger()->notice("Null id passed when processing detected references detectedContentType=\"$detectedContentType\"");
-                        } elseif (!$this->submissionManager->submissionExists($detectedContentType, $curBlogId, $detectedId, $targetBlogId)) {
+                        } elseif (!$this->submissionManager->submissionExistsNoLastError($detectedContentType, $curBlogId, $detectedId, $targetBlogId)) {
```
Comment (vsolovei-smartling) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:587]: This change allows for reupload of previously failed submissions while sending related content

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 3]
```diff
@@ -611,13 +612,11 @@ protected function normalizeReferences(array $references): array
         }
 
         if (isset($references[self::POST_BASED_PROCESSOR])) {
-            foreach (array_keys($references[self::POST_BASED_PROCESSOR]) as $postId) {
+            foreach ($references[self::POST_BASED_PROCESSOR] as $key => $value) {
+                $postId = is_int($value) ? $value : $key;
                 $postType = $this->wordpressProxy->get_post_type($postId);
-                if ($postType !== false) {
+                if (is_string($postType)) {
```
Comment (vsolovei-smartling) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:618]: The workaround: for some environments, we get unexpected string content where integers should be. I'll spend more time investigating and removing the root cause

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php [hunk 4]
```diff
@@ -611,13 +612,11 @@ protected function normalizeReferences(array $references): array
...
                     $result[$postType][] = $postId;
-                } else {
-                    $this->getLogger()->notice("Related item type couldn't be determined, assuming 'attachment', postId=$postId");
-                    $result['attachment'][] = $postId;
```
Comment (vsolovei-smartling) [inline inc/Smartling/Services/ContentRelationsDiscoveryService.php:620]: The root cause of the issue was wrong ids sent for translation, not WP unable to determine content type

**Review by vsolovei-smartling — COMMENTED** [review]
This PR has many logging changes, I believe them to be generally useful

File: readme.txt [hunk 5]
```diff
@@ -4,7 +4,7 @@ Tags: translation, localization, localisation, translate, multilingual, smartlin
 Requires at least: 5.5
 Tested up to: 6.1
 Requires PHP: 8.0
-Stable tag: 3.0.2
+Stable tag: 3.0.3
```
Comment (dimitrystd) [inline readme.txt:7]: You forgot to add text in Changelog

File: inc/Smartling/Services/ContentRelationsDiscoveryService.php (hunk 3 above)
Comment (dimitrystd) [reply inc/Smartling/Services/ContentRelationsDiscoveryService.php:618]: So why do you test for string type instead of int?

**Review by dimitrystd — APPROVED** [review]

**Pull request merged**